### PR TITLE
Wal file size bug fix

### PR DESCRIPTION
**Pull request by semil**
Fix WAL file size limit exceeded bug as described in #4290

#### Type of change

- Bug fix
- Test update

#### Related issues

#4290

### CODE REVIEW

File: vendor/go.etcd.io/etcd/server/v3/wal/decoder.go [hunk 1]
```diff
@@ -84,12 +81,15 @@ func (d *decoder) decodeRecord(rec *walpb.Record) error {
 	}
 
 	recBytes, padBytes := decodeFrameSize(l)
-	if recBytes >= maxWALEntrySizeLimit-padBytes {
-		return ErrMaxWALEntrySizeLimitExceeded
+	// The length of current WAL entry must be less than the remaining file size.
+	maxEntryLimit := fileBufReader.FileInfo().Size() - d.lastValidOff - padBytes
+	if recBytes > maxEntryLimit {
```
Comment (yacovm) [inline vendor/go.etcd.io/etcd/server/v3/wal/decoder.go:86]: This is the fix :-)

**Review by yacovm — APPROVED** [review]

**Review by denyeart — APPROVED** [review]
Thanks @semil !

**Comment by yacovm** [comment]
It failed the Raft suite again... @semil did you try running it locally?

**Comment by denyeart** [comment]
It passed the 4th time in CI, and It passed for me locally, so hopefully it is ok.
I'll keep on eye on the integration tests and see if they have in fact regressed for the worse with this change...

**Comment by denyeart** [comment]
@Mergifyio backport release-2.5

**Pull request merged**